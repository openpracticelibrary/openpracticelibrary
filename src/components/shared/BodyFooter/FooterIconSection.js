import React from "react";
import {
  Mic,
  MailOutline,
  Github,
  YouTube,
  Instagram,
} from "@material-ui/icons";
import { Grid } from "@material-ui/core";
import { Link } from "gatsby";

const sections = [
  { url: "mailto:info@openpracticelibrary.com", icon: <MailOutline /> },
  {
    url: "https://www.instagram.com/openpracticelibrary/?hl=en",
    icon: <Instagram />,
  },
  { url: "https://github.com/openpracticelibrary", icon: <Github /> },
  {
    url:
      "https://podcasts.apple.com/us/podcast/open-practice-podcast/id1501715186",
    icon: <Mic />,
  },
  {
    url:
      "https://www.youtube.com/channel/UCBGbB7xVsmf-yQ8NXn7TOVg?view_as=subscriber",
    icon: <YouTube />,
  },
];

export default function FooterIconSection(props) {
  return (
    <>
      <Grid container direction="row" spacing={2} alignItems="center">
        {sections.map((section, i) => (
          <Grid item key={i}>
            <Link href={section.url} target="_blank" rel="noopener">
              {section.icon}
            </Link>
          </Grid>
        ))}
      </Grid>
    </>
  );
}
