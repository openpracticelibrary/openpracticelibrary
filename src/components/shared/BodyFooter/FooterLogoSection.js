import React from "react";
import { Box, Typography } from "@material-ui/core";
import Logo from "../../../components/shared/Logo";
import logoLabs from "../../../assets/images/logo-red-hat-open-innovation-labs-a-standard-rgb.png";

export default function FooterLogoSection() {
  return (
    <>
      <Logo horizontal />
      <Box display="flex" alignItems="center">
        <Typography variant="overline" data-testid="poweredBy">
          powered by
        </Typography>
        <img src={logoLabs} alt="Open Innovation Labs Logo" />
      </Box>
    </>
  );
}
