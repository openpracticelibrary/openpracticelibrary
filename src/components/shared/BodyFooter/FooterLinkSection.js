import React from "react";
import { Typography, Box, Grid } from "@material-ui/core";
import FooterIconSection from "./FooterIconSection";
import { navigate } from "gatsby";

const sections = [
  { title: "Practices", url: "/practice" },
  { title: "About", url: "/about" },
  { title: "Terms", url: "https://creativecommons.org/licenses/by-sa/4.0/" },
];

export default function FooterLogoSection() {
  return (
    <>
      <Grid
        container
        direction="column"
        justify="space-between"
        alignItems="flex-end"
        spacing={2}
      >
        <Grid item>
          <Grid
            container
            direction="row"
            spacing={3}
          >
            {sections.map((section, i) => (
              <Grid item key={i}>
                <Box color="grey.A400">
                  <Typography
                    variant="button"
                    onClick={() => navigate(section.url)}
                  >
                    {section.title}
                  </Typography>
                </Box>
              </Grid>
            ))}
          </Grid>
        </Grid>
        <Grid item>
          <Grid
            container
            direction="row"
            spacing={2}
            alignItems="center"
          >
            <Grid item>
              <Box
                color="common.black"
                fontWeight={500}
              >
                <Typography
                  variant="button"
                  data-testid="connect"
                >
                  Please connect with us:
                </Typography>
              </Box>
            </Grid>
            <Grid item>
              <FooterIconSection />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </>
  );
}
