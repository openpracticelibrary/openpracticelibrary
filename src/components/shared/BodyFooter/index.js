import React from "react";
import {makeStyles} from "@material-ui/core/styles";
import FooterLogoSection from "./FooterLogoSection";
import FooterLinkSection from "./FooterLinkSection";
import { Grid, Box, Divider } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  footer: {
    padding: theme.spacing(6, 0),
  },
  trueWhiteColor: {
    backgroundColor: theme.palette.common.white,
  }
}));

export default function BodyFooter() {
  const classes = useStyles();

  return (
    <>
      <Box className={classes.trueWhiteColor}>
        <Divider/>
        <Box
          p={6}
          component="footer"
          data-testid="footer"
        >
          <Grid container>
            <Grid item xs={1}>
            </Grid>
            <Grid item xs={10}>
              <Box display="flex">
                <Box flexShrink={1}>
                  <FooterLogoSection/>
                </Box>
                <Box width="100%">
                  <FooterLinkSection/>
                </Box>
              </Box>
            </Grid>
          </Grid>
        </Box>
      </Box>
    </>
  );
}
