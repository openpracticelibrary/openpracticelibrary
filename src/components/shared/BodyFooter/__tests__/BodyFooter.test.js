import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer';
import "@testing-library/jest-dom/extend-expect";
import BodyFooter from '../index';

afterEach(cleanup);

it("renders BodyFooter without crashing", ()=>{
  const div = document.createElement("div");
  render(<BodyFooter/>, div);
});

it("renders BodyFooter correctly", ()=>{
  const { getByTestId } = render(<BodyFooter/>);
  expect(getByTestId("footer")).toBeInTheDocument();
});

it("renders children components correctly", ()=>{
  const { getByTestId } = render(<BodyFooter/>);
  expect(getByTestId("connect")).toHaveTextContent("Please connect with us:");
  expect(getByTestId("poweredBy")).toHaveTextContent("powered by");
});

it("matches the snapshot", ()=>{
  const tree = renderer.create(<BodyFooter/>).toJSON();
  expect(tree).toMatchSnapshot();
});
