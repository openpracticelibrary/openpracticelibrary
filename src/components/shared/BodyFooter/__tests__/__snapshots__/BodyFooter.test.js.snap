// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-481 makeStyles-trueWhiteColor-479"
>
  <hr
    className="MuiDivider-root"
  />
  <footer
    className="MuiBox-root MuiBox-root-489"
    data-testid="footer"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <div
          className="MuiBox-root MuiBox-root-593"
        >
          <div
            className="MuiBox-root MuiBox-root-594"
          >
            <h6
              className="MuiTypography-root MuiTypography-h6"
            >
              Open Practice Library
            </h6>
            <div
              className="MuiBox-root MuiBox-root-625"
            >
              <span
                className="MuiTypography-root MuiTypography-overline"
                data-testid="poweredBy"
              >
                powered by
              </span>
              <img
                alt="Open Innovation Labs Logo"
                src="logo-red-hat-open-innovation-labs-a-standard-rgb.png"
              />
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-626"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-space-between"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div
                  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      className="MuiBox-root MuiBox-root-627"
                    >
                      <span
                        className="MuiTypography-root MuiTypography-button"
                        onClick={[Function]}
                      >
                        Practices
                      </span>
                    </div>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      className="MuiBox-root MuiBox-root-628"
                    >
                      <span
                        className="MuiTypography-root MuiTypography-button"
                        onClick={[Function]}
                      >
                        About
                      </span>
                    </div>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      className="MuiBox-root MuiBox-root-629"
                    >
                      <span
                        className="MuiTypography-root MuiTypography-button"
                        onClick={[Function]}
                      >
                        Terms
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div
                  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      className="MuiBox-root MuiBox-root-630"
                    >
                      <span
                        className="MuiTypography-root MuiTypography-button"
                        data-testid="connect"
                      >
                        Please connect with us:
                      </span>
                    </div>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-item"
                      >
                        <a
                          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                          href="mailto:info@openpracticelibrary.com"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          target="_blank"
                        >
                          <svg
                            height="20"
                            viewBox="0 0 26 20"
                            width="26"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              fill="#717171"
                            >
                              <path
                                d="M25.713 2.794L24.707 1.79l-9.521 9.523c-1.213 1.212-3.185 1.212-4.398 0L1.267 1.789.26 2.794l7.11 7.11-7.11 7.11 1.006 1.004 7.109-7.109 1.408 1.407c.855.856 1.993 1.327 3.203 1.327 1.21 0 2.348-.471 3.203-1.327l1.408-1.407 7.11 7.11 1.005-1.005-7.11-7.11 7.11-7.11"
                              />
                              <path
                                d="M23.682 19.862H2.292C1.028 19.862 0 18.834 0 17.571V2.29C0 1.029 1.028 0 2.292 0h21.39c1.264 0 2.292 1.028 2.292 2.292V17.57c0 1.263-1.028 2.291-2.292 2.291zM2.292 1.528c-.421 0-.764.343-.764.764V17.57c0 .42.343.764.764.764h21.39c.421 0 .764-.343.764-.764V2.29c0-.42-.343-.763-.764-.763H2.292z"
                              />
                            </g>
                          </svg>
                        </a>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item"
                      >
                        <a
                          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                          href="https://www.instagram.com/openpracticelibrary/?hl=en"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          target="_blank"
                        >
                          <svg
                            height="22"
                            viewBox="0 0 22 22"
                            width="22"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              fill="#717171"
                              fillRule="evenodd"
                            >
                              <path
                                d="M10.974.007c-2.98 0-3.354.013-4.525.066-1.168.053-1.965.239-2.663.51-.722.28-1.334.656-1.944 1.266-.61.61-.985 1.222-1.266 1.943C.305 4.49.12 5.288.066 6.456.013 7.626 0 8 0 10.981c0 2.98.013 3.354.066 4.524.053 1.168.239 1.966.51 2.664.28.722.656 1.334 1.266 1.944.61.61 1.222.985 1.944 1.265.698.272 1.495.457 2.663.51 1.17.054 1.545.067 4.525.067 2.98 0 3.354-.013 4.524-.066 1.168-.054 1.966-.24 2.664-.51.722-.281 1.334-.656 1.944-1.266.61-.61.985-1.222 1.266-1.944.27-.698.456-1.496.51-2.664.053-1.17.066-1.544.066-4.524s-.013-3.354-.066-4.525c-.054-1.168-.24-1.966-.51-2.664-.28-.721-.656-1.333-1.266-1.943-.61-.61-1.222-.986-1.944-1.266-.698-.271-1.496-.457-2.664-.51-1.17-.053-1.544-.066-4.524-.066zm0 1.977c2.93 0 3.277.011 4.434.064 1.07.049 1.651.228 2.038.378.512.2.878.437 1.262.82.384.385.621.75.82 1.263.15.386.33.967.378 2.037.053 1.158.064 1.505.064 4.435 0 2.93-.01 3.277-.064 4.434-.048 1.07-.227 1.651-.377 2.038-.2.512-.437.878-.821 1.262-.384.384-.75.621-1.262.82-.387.15-.968.33-2.038.378-1.157.053-1.504.064-4.434.064s-3.278-.01-4.435-.064c-1.07-.048-1.65-.227-2.037-.378-.512-.199-.878-.436-1.262-.82-.384-.384-.622-.75-.82-1.262-.151-.387-.33-.968-.379-2.038-.053-1.157-.064-1.504-.064-4.434s.011-3.277.064-4.435c.049-1.07.228-1.65.378-2.037.2-.513.437-.878.821-1.262.384-.384.75-.622 1.262-.821.386-.15.968-.329 2.037-.378 1.158-.053 1.505-.064 4.435-.064z"
                              />
                              <path
                                d="M10.974 14.639c-2.02 0-3.658-1.638-3.658-3.658s1.638-3.658 3.658-3.658 3.658 1.638 3.658 3.658-1.638 3.658-3.658 3.658zm0-9.293c-3.112 0-5.635 2.522-5.635 5.635 0 3.112 2.523 5.635 5.635 5.635 3.112 0 5.635-2.523 5.635-5.635 0-3.113-2.523-5.635-5.635-5.635zM18.149 5.123c0 .727-.59 1.317-1.317 1.317-.728 0-1.317-.59-1.317-1.317s.59-1.317 1.317-1.317 1.317.59 1.317 1.317"
                              />
                            </g>
                          </svg>
                        </a>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item"
                      >
                        <a
                          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                          href="https://github.com/openpracticelibrary"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          target="_blank"
                        >
                          <svg
                            height="22"
                            viewBox="0 0 23 22"
                            width="23"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M11.277 0C5.05 0 0 5.05 0 11.279c0 4.983 3.231 9.21 7.713 10.701.564.104.77-.244.77-.543 0-.268-.01-.977-.015-1.918-3.137.68-3.8-1.513-3.8-1.513-.512-1.303-1.252-1.65-1.252-1.65-1.024-.699.078-.685.078-.685 1.132.08 1.727 1.163 1.727 1.163 1.006 1.723 2.64 1.225 3.283.936.102-.728.394-1.225.716-1.507-2.505-.284-5.138-1.252-5.138-5.574 0-1.231.44-2.239 1.161-3.027-.116-.285-.503-1.432.111-2.984 0 0 .947-.304 3.101 1.156.9-.25 1.865-.375 2.824-.38.958.005 1.922.13 2.823.38 2.153-1.46 3.099-1.156 3.099-1.156.615 1.552.228 2.7.112 2.984.723.788 1.16 1.796 1.16 3.027 0 4.333-2.638 5.286-5.15 5.565.405.348.765 1.037.765 2.089 0 1.507-.013 2.724-.013 3.094 0 .301.203.652.775.542 4.478-1.495 7.707-5.719 7.707-10.7C22.557 5.049 17.507 0 11.277 0"
                              fill="#717171"
                              fillRule="evenodd"
                            />
                          </svg>
                        </a>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item"
                      >
                        <a
                          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                          href="https://podcasts.apple.com/us/podcast/open-practice-podcast/id1501715186"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          target="_blank"
                        >
                          <svg
                            height="22"
                            viewBox="0 0 22 22"
                            width="22"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M10.997 0C4.93 0 0 4.936 0 11.003c0 4.374 2.593 8.336 6.602 10.084.227.102.49.075.692-.071.2-.147.307-.39.278-.637-.028-.246-.187-.459-.416-.556-3.512-1.53-5.781-4.99-5.781-8.82 0-5.324 4.298-9.628 9.622-9.628s9.628 4.304 9.628 9.628c0 3.828-2.266 7.289-5.775 8.82-.233.094-.396.307-.426.556-.03.25.078.495.281.641.204.146.472.17.698.062C19.41 19.332 22 15.374 22 11.002 22 4.937 17.064 0 10.997 0zm0 4.385c-3.646 0-6.611 2.971-6.611 6.618 0 1.82.746 3.561 2.07 4.81.276.262.712.249.972-.027.26-.277.248-.712-.028-.973-1.049-.99-1.64-2.368-1.64-3.81 0-2.904 2.334-5.243 5.237-5.243 2.904 0 5.242 2.34 5.242 5.243 0 1.443-.593 2.82-1.643 3.81-.179.17-.254.42-.197.66.056.238.236.43.472.5.235.07.49.01.669-.16 1.325-1.25 2.076-2.989 2.076-4.81 0-3.647-2.972-6.618-6.619-6.618zm0 3.46c-1.496 0-2.72 1.225-2.72 2.721s1.224 2.726 2.72 2.726c1.496 0 2.726-1.23 2.726-2.726s-1.23-2.72-2.726-2.72zm0 6.141c-1.511 0-2.744 1.24-2.744 2.752v2.512c0 1.512 1.233 2.75 2.744 2.75 1.512 0 2.752-1.238 2.752-2.75v-2.512c0-1.512-1.24-2.752-2.752-2.752z"
                              fill="#717171"
                            />
                          </svg>
                        </a>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item"
                      >
                        <a
                          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                          href="https://www.youtube.com/channel/UCBGbB7xVsmf-yQ8NXn7TOVg?view_as=subscriber"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          target="_blank"
                        >
                          <svg
                            height="20"
                            viewBox="0 0 29 20"
                            width="29"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              fill="none"
                              fillRule="evenodd"
                            >
                              <path
                                d="M27.662 3.088C27.337 1.873 26.38.915 25.164.59 22.961 0 14.126 0 14.126 0S5.291 0 3.088.59C1.873.915.915 1.873.59 3.088 0 5.29 0 9.888 0 9.888s0 4.597.59 6.8c.325 1.216 1.283 2.173 2.498 2.498 2.203.59 11.038.59 11.038.59s8.835 0 11.038-.59c1.216-.325 2.173-1.282 2.498-2.498.59-2.203.59-6.8.59-6.8s0-4.597-.59-6.8"
                                fill="#717171"
                              />
                              <path
                                d="M11.301 14.126L18.641 9.888 11.301 5.65 11.301 14.126"
                                fill="#FFFFFE"
                              />
                            </g>
                          </svg>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
